guides/doc-Upgrading_and_Updating/topics/upgrading_capsule_server.adoc

```diff
@@ -41,20 +41,41 @@ endif::[]
 For information on backups, see {AdministeringDocURL}Backing_Up_Server_and_Proxy_admin[Backing Up {ProjectServer} and {SmartProxyServer}] in the _Administering {ProjectName} {ProjectVersionPrevious}_ guide.
 
 ifdef::katello,satellite[]
-+
 . Regenerate certificates.
 On the main {Project} server:
 +
 [options="nowrap" subs="attributes"]
 ----
-# {certs-generate} --foreman-proxy-fqdn "myproxy.example.com" \
+# {certs-generate} --foreman-proxy-fqdn "{smartproxy-example-com}" \
                        --certs-update-all \
-                       --certs-tar "~/myproxy.example.com-certs.tar"
+                       --certs-tar "~/{smartproxy-example-com}-certs.tar"
 ----
 +
-For more information on custom SSL certificates signed by a Certificate Authority, see {InstallingSmartProxyDocURL}deploying-a-custom-ssl-certificate-to-{smart-proxy-context}-server_{smart-proxy-context}[Deploying a Custom SSL Certificate to {SmartProxyServer}] in _{InstallingSmartProxyDocTitle}_.
+
+. If you use custom SSL certificates on your {smart-proxy-context} server, use the following command to regenerate the certificate tar file on the {ProjectServer}:
 +
-. Copy the resulting tarball to your {SmartProxy}, for this example we will use `/root/myproxy.example.com-certs.tar`
+[options="nowrap" subs="attributes"]
+----
+# {certs-generate} \
+--foreman-proxy-fqdn "{smartproxy-example-com}" \
+--certs-tar "~/{smartproxy-example-com}-certs.tar" \
+--server-cert "/path/to/{smart-proxy-context}_cert.pem" \
+--server-key "/path/to/{smart-proxy-context}_key.pem" \
+--server-ca-cert "/path/to/{smart-proxy-context}_ca.pem" \
+--certs-update-server
+----
++
+For more information on custom SSL certificates signed by a Certificate Authority, see {InstallingSmartProxyDocURL}deploying-a-custom-ssl-certificate-to-{smart-proxy-context}-server_{smart-proxy-context}[Deploying a Custom SSL Certificate to {smart-proxy-context}] in _{InstallingSmartProxyDocTitle}_.
+
+. Copy the resulting tarball to your {SmartProxy}, for this example we will use `/root/{smartproxy-example-com}-certs.tar`
+
+. Run the installer:
++
+[options="nowrap" subs="attributes"]
+----
+# {foreman-installer} --certs-tar-file /root/{smartproxy-example-com}-certs.tar \
+--certs-update-all --certs-regenerate true --certs-deploy true
+----
```

Comment:
I now further read this and this step is redundant, at least in the Katello case. You already run it lower, after `yum -y update`.

I'm not sure what happens in the downstream case. There it may be needed, or place it in the file the answers already exists.

Apologies for the confusion and back and forth on this, but I think this is really the result of a very vague PR description. You don't explain what you are doing and why. That means as a reviewer I'm having to guess and that causes problems. That is why we always insist on clear descriptions of PRs. It makes the review process easier.

Reply:
> I now further read this and this step is redundant, at least in the Katello case. You already run it lower, after `yum -y update`.
> 
> I'm not sure what happens in the downstream case. There it may be needed, or place it in the file the answers already exists.
> 
> Apologies for the confusion and back and forth on this, but I think this is really the result of a very vague PR description. You don't explain what you are doing and why. That means as a reviewer I'm having to guess and that causes problems. That is why we always insist on clear descriptions of PRs. It makes the review process easier.

Apologies for the confusion in PR msg, but kindly make it clear what changes do you suggest now. Should we remove this step or any other update is needed? It would have been very easier if this gets clear in the BZ itself. We have changed it a lot since we made the original commit as per the discussion in the BZ mentioned in the original commit msg. 

Thanks!

Reply:
@ekohl, gentle reminder!

Reply:
> It would have been very easier if this gets clear in the BZ itself.

This is a much deeper issue, but I think the problem here is that the experts on the matter are not looking at the BZs. Even worse, the BZ is full of private comments. It is a very downstream focused way of working. In the ideal world all a reviewer needs is this single PR and has all the information. As the PR author that is your job.

Because I thought it was easier to demonstrate using code, I'd have expected something like https://github.com/theforeman/foreman-documentation/pull/1499. That has the context of why it splits the instructions. I also think it is the most minimal solution to the problem, though the phrasing can probably be improved.

Reply:
> > It would have been very easier if this gets clear in the BZ itself.
> 
> This is a much deeper issue, but I think the problem here is that the experts on the matter are not looking at the BZs. Even worse, the BZ is full of private comments. It is a very downstream focused way of working. In the ideal world all a reviewer needs is this single PR and has all the information. As the PR author that is your job.

The BZ has private comments and there is no essential information in it which would hamper the working of upstream. The private comments are added for reason and for internal use only. In most cases, they are requests and information on reassigning tasks. 
 
> Because I thought it was easier to demonstrate using code, I'd have expected something like #1499. That has the context of why it splits the instructions. I also think it is the most minimal solution to the problem, though the phrasing can probably be improved.

What I have observed is approach has been changed over a period of time spent in working of this BZ. Which the writer absolutely has no control over. Also, updates are being suggested for the part which is not added or updated as a part of this PR. This makes the PR more complected. 

If PR #1499 having all the changes in the way it is expected, kindly inform if this PR can be closed without merge and update the BZ accordingly.

Reply:
> What I have observed in this BZ is, approach has been changed over a period of time spent in working of this BZ. Which the writer absolutely has no control over. Also, updates are being suggested for the part which is not added or updated as a part of this PR. This makes the PR more complected.

That's true, mostly because the BZ also didn't consider that upstream and downstream have different procedures, leading to more confusion. The described problem only exists in the downstream procedure while I was looking at it and seeing you modify the upstream procedure.

And I do agree that the scope was being increased. However, it's perfectly fine to postpone that work to a later PR. That's also what I did in #1499. Just because someone asks something, doesn't mean you have to do it. It's always a discussion.

To be more constructive: if in the future anything regarding the installer or platform is unclear, please ask me or anyone else from the platform team. I'd be happy to take a look early in the process. Hopefully we can remove a lot of the back and forth then.

> Suggestions are not provided all at once. They are provided in parts. This makes the PR more complected.

That's true. As I said, that often comes from an unclear problem description in the PR so it's hard to verify you're actually achieving what you intend to.

> If PR https://github.com/theforeman/foreman-documentation/pull/1499 having all the changes in the way it is expected, kindly inform if this PR can be closed without merge and update the BZ accordingly.

I think it is. On a technical level I think it solves the issue, but I'm not a writer so I'd appreciate a critical eye on that.

Reply:
> > Suggestions are not provided all at once. They are provided in parts. This makes the PR more complected.
> 
> That's true. As I said, that often comes from an unclear problem description in the PR so it's hard to verify you're actually achieving what you intend to.
Unclear PR description does not mean you should review only half of the content and leave rest of the part. :)
> 
> > If PR #1499 having all the changes in the way it is expected, kindly inform if this PR can be closed without merge and update the BZ accordingly.
> 
> I think it is. On a technical level I think it solves the issue, but I'm not a writer so I'd appreciate a critical eye on that.
Thank you for confirming that this PR can be closed. I am closing this one and you may continue merging the other PR when its ready.